### PR TITLE
incremental build: add check for commit sha

### PR DESCRIPTION
**Pull request by ansyral**
the format of the changesfile would be like:
```
<from>	12345
<to>	56789
index.md	Updated
```
and `from`, `to` are optional.
@herohua @vwxyzh @hellosnow @superyyrrzz @qinezh @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 1]
```diff
@@ -44,14 +44,18 @@ public class DocumentBuilder : IDisposable
             IEnumerable<Assembly> assemblies,
             ImmutableArray<string> postProcessorNames,
             string templateHash,
-            string intermediateFolder = null)
+            string intermediateFolder = null,
+            string commitFromSHA = null,
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: `SHA` -> `Sha`?

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs (hunk 1 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: in camel casing should use `Sha`? 
[http://dotnet.github.io/docfx/guideline/csharp_coding_standards.html#naming-mandatory](http://dotnet.github.io/docfx/guideline/csharp_coding_standards.html#naming-mandatory)

---
In reply to: [85288594](https://github.com/dotnet/docfx/pull/831#discussion_r85288594) [](ancestors = 85288594)

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**